### ISSUE TITLE
Cannot upload files + text fields from superagent

### ISSUE THREAD

**Issue by DarianAnjuhal** [expressjs/multer#212]
Hey,

multer is really nice, and I hopefully we can use it in the future too.

I am currently working an a ContentServer and I wanna upload Images/Videos/... from a lot of different points (Browser, nodejs, nodejs-tests,...etc)

Additionally its important to add infos to the upload.
Normally it uploads the files but I cannot get the infos. Sometimes it works, sometimes it doesnt.

Code for Tests using superagent:

```javascript
request("http://localhost:3000")
.post("/testupload")
      .attach('image', '/path/to/file')
      .field('content_group', "teststuff")
      .field('use_timestamp', "false")
      .field('generate_thumbnail', "false")
      .field('OPTIONS', "XXXXXXXXXXXXXXXX")
      .send()
      
      .end(function(err,res) {
      if (err) {
        throw err;
      }
```

Current Server application:

```javascript
var express = require('express');
var multer  = require('multer');

var app = express();
 
app.post('/testupload', function (req, res, next) {

  var upload = multer({ dest: 'uploads/'}).single('image');
  
  //<-- programm always comes here 
  upload(req, res, function (err) {

    //<-- programm very often doesnt come here

    if (err) {
      console.log(err);
   
    }

    //try to get fields and file infos
    console.log(req.file);
    console.log(req.body);

    res.header("Access-Control-Allow-Origin", "*");
    res.sendStatus(200);
  });
});

app.options('/testupload', function (req, res, next) {

  res.header("Access-Control-Allow-Origin", "*");
  res.sendStatus(200);
});

app.listen(3000, function() {
  console.log("Server is running");
});
```

I compared every upload software with http://www.posttestserver.com and it should be totally the same but multer does react different.

Is this a known issue or what I am doing wrong? Would be nice if someone can help me out. :-)

Thx and bye
Darian

**Comment by siyb** [comment]
might be related to: https://github.com/expressjs/multer/pull/205

**Comment by BigstickCarpet** [comment]
I'm seeing the same behavior.  It doesn't happen with Multer 0.x.  Only the new 1.x versions.

**Comment by kadishmal** [comment]
Here is how I do. Hope it helps. Today upgraded `multer` to 1.0.3 and it works.

	request(app)
			.post(postURI)
			.set({
				'Accept': 'application/json',
				'Authorization': 'Bearer ' + accessToken
			})
			.field('version', ndkSymbolInfoLibGame.version)
			.field('platform', ndkSymbolInfoLibGame.platform)
			.attach('file', ndkSymbolInfoLibGame.file)
			.expect({
				status: 400,
				message: 'The file size exceeded the maximum size permitted (Max: ' + conf.maxUploadFileSize + ' bytes).'
			})
			.expect(400, done);

**Comment by BigstickCarpet** [comment]
hmmm.... my code is basically the same as yours, but mine doesn't work.  What version of Node are you using?  I'm on v0.12.

**Comment by kadishmal** [comment]
I use v0.10.38.

**Comment by kadishmal** [comment]
Note that I don't call `send` after `attach`. I don't know if it affects, can't test now.

**Comment by LinusU** [comment]
This should be fixed by #205 and released on npm as version `1.0.5` :rocket: 

Please reopen if issue persists.
